File: src/main/java/com/kerimsenturk/labreport/model/Report.java
package com.kerimsenturk.labreport.model;

import com.kerimsenturk.labreport.model.enums.ReportType;
import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name = "reports")
public class Report {
    @Id
    @Column(name = "report_id")
    String reportId;
    @Column(name = "title")
    String title;
    @Column(name = "details")
    String details;
    @Column(name = "issue_date")
    Date issueDate;
    @Column(name = "file_path")
    String filePath;
    @Enumerated(EnumType.STRING)
    @Column(name = "report_type")
    ReportType reportType;

    public Report() {
    }

    public Report(String reportId, String title, String details, Date issueDate, String filePath, ReportType reportType) {
        this.reportId = reportId;
        this.title = title;
        this.details = details;
        this.issueDate = issueDate;
        this.filePath = filePath;
        this.reportType = reportType;
    }

    public String getReportId() {
        return reportId;
    }

    public void setReportId(String reportId) {
        this.reportId = reportId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    public Date getIssueDate() {
        return issueDate;
    }

    public void setIssueDate(Date issueDate) {
        this.issueDate = issueDate;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public ReportType getReportType() {
        return reportType;
    }

    public void setReportType(ReportType reportType) {
        this.reportType = reportType;
    }
}

File: src/main/web-ui/src/domain/model/Report.ts

export enum ReportType{
    PATHOLOGICAL = "PATHOLOGICAL",
    DIAGNOSTIC = "DIAGNOSTIC"
}

export class Report {
    private _reportId: string;
    private _title: string;
    private _details: string;
    private _issueDate: Date;
    private _filePath: string;
    private _reportType: ReportType;

    constructor(reportId: string, title: string, details: string,
                issueDate: Date, filePath: string, reportType: ReportType) {
        this._reportId = reportId;
        this._title = title;
        this._details = details;
        this._issueDate = issueDate;
        this._filePath = filePath;
        this._reportType = reportType;
    }

    get reportId(): string {
        return this._reportId;
    }

    set reportId(value: string) {
        this._reportId = value;
    }

    get title(): string {
        return this._title;
    }

    set title(value: string) {
        this._title = value;
    }

    get details(): string {
        return this._details;
    }

    set details(value: string) {
        this._details = value;
    }

    get issueDate(): Date {
        return this._issueDate;
    }

    set issueDate(value: Date) {
        this._issueDate = value;
    }

    get filePath(): string {
        return this._filePath;
    }

    set filePath(value: string) {
        this._filePath = value;
    }

    get reportType(): ReportType {
        return this._reportType;
    }

    set reportType(value: ReportType) {
        this._reportType = value;
    }
}
